### DISCUSSION TITLE
Accessing orignal url in middleware, when having a default locale

### DISCUSSION THREAD

**Discussion by stijlfigurant** [vercel/next.js#35180] [Help]
I have a default locale: "en", and I have a middleware 

In the middleware I need to check if a certain locale part is present in the url. 

When I am reading `req.url` in my middleware, im getting the url, but with the default locale stripped from it.

So say; http://localhost:3000/en/my-cool-path

When reading `req.url`, i get http://localhost:3000/my-cool-path, but the /en bit omitted. 

I understand its accessible in `req.nextUrl.locale` but I can't trust that as I also have cases of deprecated locales that then default to req.nextUrl.locale = en; but that case shouldn't be handled in my middleware.

In short, I just would like `req.url` to output the exact url as it is in my browser so I can do with it as I please in my middleware. The whole magical omitting the default locale part is not helping. 

Please advise.

**Comment by Thamasky** [comment]
Hey there,

Sorry for not giving you a direct answer to your question, but the following might help.
I don't know exactly the use case you want to support, but based on your description I think you'd like to use the following workaround: [Prefixing the default locale](https://nextjs.org/docs/advanced-features/i18n-routing#prefixing-the-default-locale)

Using that workaround, you can make it so the default locale doesn't get omitted anymore. So if your default locale is `en`, it will always be present in the url.

**Comment by nino1337** [comment]
See my original [comment](https://github.com/vercel/next.js/discussions/50575) for the solution.
